### PR TITLE
Stringify Sidekiq.default_worker_options's keys, fix #2124

### LINKED ISSUE #2124: default_worker_options method doesn't use stringify_keys
When defining worker options in a worker, the [sidekiq_options](https://github.com/mperham/sidekiq/blob/27fc88504b15ea5b57929845c7118de66c3120ea/lib/sidekiq/worker.rb#L67) method uses `stringify_keys` to ensure the options all have strings as keys.

Unfortunately the [default_worker_options](https://github.com/mperham/sidekiq/blob/c4fc6dad68f14053af445a52e7e141c312ae6d13/lib/sidekiq.rb#L100) method doesn't stringify the keys. This can lead to conflicting options for a worker:

```ruby
pry(main)> EmailFetcherJob.get_sidekiq_options
=> {"retry"=>false, "queue"=>"default", :retry=>0, :backtrace=>true}
```

### PR DESCRIPTION
**Pull request by seuros**

### CODE REVIEW

File: Changes.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 HEAD
 -----------
+- Stringify Sidekiq.default_worker_options's keys [#]
```
Comment (mperham) [inline Changes.md]: Missing issue #?

File: lib/sidekiq.rb [hunk 2]
```diff
@@ -97,7 +97,7 @@ def self.server_middleware
   end
 
   def self.default_worker_options=(hash)
-    @default_worker_options = default_worker_options.merge(hash)
+    @default_worker_options = default_worker_options.merge(hash).stringify_keys!
```
Comment (mperham) [inline lib/sidekiq.rb]: I believe that stringify_keys! is Rails-specific?  It needs to work with plain Ruby.

Comment (seuros) [inline lib/sidekiq.rb]: That right. Fixing it.

File: Changes.md (hunk 1 above)
Comment (seuros) [inline Changes.md]: Fixed

File: lib/sidekiq.rb [hunk 3]
```diff
@@ -97,7 +97,7 @@ def self.server_middleware
   end
 
   def self.default_worker_options=(hash)
-    @default_worker_options = default_worker_options.merge(hash)
+    @default_worker_options = default_worker_options.merge(hash).stringify_keys
```
Comment (mperham) [inline lib/sidekiq.rb]: You need to stringify the hash, not the result.

```
> { 'a' => 1 }.merge(a: 2).stringify_keys
NoMethodError: undefined method `stringify_keys' for {"a"=>1, :a=>2}:Hash
```

Comment (seuros) [inline lib/sidekiq.rb]: require 'sidekiq/core_ext' first

**Comment by mperham** [comment]
Perfect, thanks!

**Pull request merged**